site/source/tutorials/1040_data_driven_widgets/index.md

```diff
@@ -0,0 +1,193 @@
+
+The next step to update the `Banner` widget to pass the correct `data` into the `List`.
+
+{% task 'Pass worker data down through the `Banner` to the `List`' %}
+
+Currently, the application keeps all worker data as a private variable within the `App` widget. In order to use this data to also power the `List`, the `Banner` widget must also be updated to accept a `data` property so it can in turn pass it down to the `List` it renders.
+
+{% instruction 'Export a `BannerProperties` interface from the `Banner` widget so a `data` array property can be passed in:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Banner.ts' lines:4,5-8 %}
+
+{% instruction 'Update the generic parameter passed into `WidgetBase` to use the new interface:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Banner.ts' lines:10 %}
+
+{% instruction 'Pass worker data to the routing outlet for the `Banner` widget in `App.ts`:' %}
+
+```ts
+w(BannerOutlet, {
+    data: this._workerData
+})
+```
+
+Now that the `Banner` widget has worker data, it can pass a subset of this data into the `List` widget based on the current `List` input value.
+
+{% instruction 'Pass a subset of worker data into the `List` based on its `value` in `Banner.ts`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Banner.ts' lines:11-35 %}
+
+A helper method was added that filters worker data items by name based on a query value. These filtered items are passed to the `List` as its `data`, causing it to render result items if any exist based on user input.
+
+{% section %}
+
+## Summary
+
+A filterable list widget was built throughout this tutorial to demonstrate how Dojo 2 widgets have no coupling to a specific store implementation. Despite the simplicity of this example widget, it demonstrates an key departure from Dojo 1 and other widget frameworks that lack reactivity: data-driven widgets request data from their parent and parents pass data back down to their children.
+
+It's also important to note that in this tutorial, hardcoded worker data is passed down from the `App` widget to the `Banner` widget and ultimately to the `List` widget, but the data could've come from anywhere, including a remote server. It's entirely possible for `Banner` to directly initiate an XHR request for data, and it's also possible for `Banner` to dispatch an action to an application store that in turn requests data. The flexibility provided by the widget system makes it so the `List` doesn't care _where_ its data comes from. Instead, data-driven widgets in Dojo 2 both request and receive data using their `properties`, and the parent is responsible for either relaying or initiating the data request itself.
```

Comment:
Regarding the abrupt ending, you could add some asides about styling (which would link to the work in progress styling tutorial, so that could be coming soon for now), and for events, remind them of https://dojo.io/tutorials/004_user_interactions/ if they want to add events to the items. Or you could quickly add event handlers and refer to that tutorial as a refresher. We definitely don't want to really repeat all concepts in every tutorial, so it's better to err on the side of less in this case. We may want to eventually have some cookbook style tutorials that are complete demos from start to finish, but I would say that's not what we're aiming for here.

Reply:
That said, it probably wouldn't hurt to just provide some default styling throughout so the example looks good maybe, and just provide that with as minimal of an impact as possible on the tutorial?